### PR TITLE
Backport Number platform from Core

### PR DESCRIPTION
**Pull request by tetienne**
See https://github.com/home-assistant/core/blob/dev/homeassistant/components/overkiz/number.py

<a href="https://gitpod.io/#https://github.com/iMicknl/ha-tahoma/pull/744"></a>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8, 3.9]
+        python-version: [3.9, 3.10]
```
Comment (tetienne) [inline .github/workflows/test.yml:11]: Core only run the test on 3.9: https://github.com/home-assistant/core/blob/dev/.github/workflows/ci.yaml#L723

File: custom_components/tahoma/number.py [hunk 2]
```diff
@@ -57,63 +75,53 @@
...
+    entities: list[OverkizNumber] = []
 
-    entities = []
-
-    key_supported_states = {
-        description.key: description for description in NUMBER_DESCRIPTIONS
-    }
-
-    for device in coordinator.data.values():
+    for device in data.coordinator.data.values():
```
Comment (tetienne) [inline custom_components/tahoma/number.py:90]: @iMicknl Coordinator is now a property of data?

File: custom_components/tahoma/cover.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 from pyoverkiz.enums import UIClass
 import voluptuous as vol
 
+from custom_components.tahoma import HomeAssistantOverkizData
```
Comment (iMicknl) [inline custom_components/tahoma/cover.py:11]: Should we just import from `.` here? (just like we do in core).

Comment (tetienne) [reply custom_components/tahoma/cover.py:11]: I will mimic what we do on the Core side 👍

File: custom_components/tahoma/__init__.py [hunk 4]
```diff
@@ -42,6 +44,15 @@
 SERVICE_EXECUTE_COMMAND = "execute_command"
 
 
+@dataclass
+class HomeAssistantOverkizData:
```
Comment (tetienne) [inline custom_components/tahoma/__init__.py:48]: Was a nice addition, avoid the magic string.

Comment (iMicknl) [reply custom_components/tahoma/__init__.py:48]: yeah I like it, took it from Tuya integration

File: requirements_dev.txt [hunk 5]
```diff
@@ -1,2 +1,2 @@
 -r requirements.txt
-homeassistant==2021.12.0b6
+homeassistant==2022.2.0b6
```
Comment (iMicknl) [inline requirements_dev.txt:2]: Can you bump required version for HACS as well?

File: requirements_test.txt [hunk 6]
```diff
@@ -1,5 +1,5 @@
 -r requirements_dev.txt
-pytest-homeassistant-custom-component==0.5.3
+pytest-homeassistant-custom-component==0.6.5
```
Comment (iMicknl) [inline requirements_test.txt:2]:
```suggestion
pytest-homeassistant-custom-component==0.6.6
```

File: requirements_dev.txt (hunk 5 above)
Comment (iMicknl) [inline requirements_dev.txt:2]:
```suggestion
homeassistant==2022.2.0
```

File: requirements_test.txt (hunk 6 above)
Comment (tetienne) [reply requirements_test.txt:2]: image
Meh

**Review by iMicknl — APPROVED** [review]
LGTM! (when tested locally)

**Comment by tetienne** [comment]
@iMicknl I tested locally

**Pull request merged**